spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+include SessionsHelper
+
+RSpec.describe UsersController, type: :controller do
+  
+  before do
+    Rails.application.load_seed
+  end
+
+  let(:user) { User.find_by(username: 'user1') }
+
+  context 'GET #new' do
+    it 'returns http success' do
+      get :new
+      expect(response).to have_http_status(:success)
+    end
+  end
+  context 'GET #show' do
+    it 'returns http success' do
+      get :show, params: { id: 1 }
+      expect(response).to have_http_status(:success)
+    end
+  end
+  context 'GET #edit' do
+    context 'when user tries to access the wrong profile' do
+      it 'redirects user to their own profile' do
+        log_in(User.first)
+        get :edit, params: { id: 2 } # 'id: 2' is the wrong user
+        expect(response).to redirect_to(User.first)
+      end
+    end
+
+    context 'when correct user accesses edit page' do
+      it 'returns http success' do
+        log_in(User.first)
+        get :edit, params: { id: 1 }
+        expect(response).to have_http_status(:success)
+      end
+    end
+  end
+
+  context 'PATCH #update' do
+    context 'when user saves an edit to their profile' do
```

Comment:
https://github.com/PolarisTLX/members-only/pull/1/files/1267e09954ae1e53721b4ded38677f7b27f560f2#r188770491

Answering your question, here is a quote of the error:

> No route matches {:action=>"/users/1", :controller=>"users", :user=>{:username=>"NewName", :password=>"password", :password_confirmation=>"password"}}

You are passing a path instead of action:) Try this:

```
it 'properly saves the changes' do
        log_in(User.first)
        patch :update, params: { id: User.first.id, user: { username: 'NewName', password: 'password', password_confirmation: 'password' } }
        expect(User.first.username).to eq('NewName')
      end
```